src/main/java/reposense/model/AuthorConfiguration.java

```diff
@@ -145,10 +145,27 @@ public void addAuthor(Author author, List<String> ignoreGlobList) {
     public void removeIgnoredAuthors(List<String> ignoredAuthorsList) {
         for (String author : ignoredAuthorsList) {
             if (authorEmailsAndAliasesMap.containsKey(author)) {
-                authorList.remove(authorEmailsAndAliasesMap.get(author));
+                removeAuthor(author);
             }
         }
-        resetAuthorInformation();
+    }
+
+    /**
+     * Removes all information of the {@code author} from the configs
+     * Precondition: {@code author} must be present in {@code authorEmailsAndAliasesMap}
+     * @param author Can be an author's git ID, email, or alias
+     */
+    public void removeAuthor(String author) {
+        Author authorToRemove = authorEmailsAndAliasesMap.get(author);
+        authorList.remove(authorToRemove);
+        authorDisplayNameMap.remove(authorToRemove);
+        authorEmailsAndAliasesMap.remove(authorToRemove.getGitId());
+
+        List<String> aliases = authorToRemove.getAuthorAliases();
+        aliases.forEach(alias -> authorEmailsAndAliasesMap.remove(alias));
```

Comment:
why is there a need to remove both alias and emails in the map? are the key value pairs not always `email:alias`?

Reply:
Nope. The key values are: `alias`, `emails` and `gitID`

Reply:
Then the variable name does not reflect the underlying structure of the map? According to the name, it feels like it should be `email:alias` instead. Perhaps can change the variable name to reflect the key -> value pairs?

Reply:
If referring to `authorEmailsAndAliasesMap`, it's the one used everywhere. I have just continued the convention. It's a pre-defined Map.

Reply:
Perhaps can create an issue for the inaccurate naming of variable `authorEmailsAndAliasesMap`, since the underlying structure is:
```
git id -> author
email -> author
alias -> author
```